[OSDOCS-4294]: adding new mce overview topic (#52801)
openshift/openshift-docs: architecture/mce-overview-ocp.adoc

:_content-type: ASSEMBLY
[id="mce-overview-ocp"]
= About multicluster engine for Kubernetes operator
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: mce-overview-ocp

toc::[]

> One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator. The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available as part of Red Hat Advanced Cluster Management for Kubernetes, or as a standalone operator that is supported as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription.

Reviewer: ```suggestion
One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator. The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available as a standalone operator that is supported as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription or as part of Red Hat Advanced Cluster Management for Kubernetes.
```

Reviewer: This is OpenShift documentation, makes sense to start with OpenShift first IMO.

> One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator. The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available as part of Red Hat Advanced Cluster Management for Kubernetes, or as a standalone operator that is supported as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription.

Author: @julienlim Per the product naming guidelines, we don't use the MCE abbreviation in the product documentation.

> One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator. The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available as part of Red Hat Advanced Cluster Management for Kubernetes, or as a standalone operator that is supported as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription.

Reviewer 2: We got legal approval (earlier in this year) from RH naming and brand to use MCE after it has been introduced. Brandi / Chris can confirm.

> One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator. The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available as part of Red Hat Advanced Cluster Management for Kubernetes, or as a standalone operator that is supported as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription.

Author: @berenss Thanks. I'm re-confirming with Brandi. While I was working on my first draft of this a couple weeks ago, I had chatted with her about whether we use "MCE" in the docs, and she said that we don't.

> One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator. The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available as part of Red Hat Advanced Cluster Management for Kubernetes, or as a standalone operator that is supported as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription.

Author: Brandi just confirmed again that we don't abbreviate MCE in the documentation.

> If you use Red Hat Advanced Cluster Management, multicluster engine is included and provides the following capabilities:

Reviewer: ```suggestion
The multicluster engine provides the following capabilities:
```

Adding "If you use RHACM ..." is confusing, brings the reader back to asking "oh I need ACM, hmmm..."

Reviewer 2: agree

> 

Reviewer 2: I would make 2 sentences:
"If you use Red Hat Advanced Cluster Management, the multicluster engine is already included as a prerequisite. 
The multicluster engine provides the following capabilities:"

Reviewer 3: Can we add a hyperlink to RHACM for the RHACM mention?

* klusterlet agent, which registers managed clusters to the hub.
* Infrastructure Operator, which manages the deployment of the Assisted Service to orchestrate on-premise bare metal and vSphere installations of OpenShift, such as SNO on bare metal. The Infrastructure Operator includes Central Infrastructure Management (CIM), which provides an interactive user interface in the Red Hat Advanced Cluster Management console where you can provision clusters.
* Open cluster management, which provides resources to manage multiple Kubernetes clusters.

> If you use {product-title} 4.8.2 or later, you can separately install multicluster engine to access the following capabilities:

Reviewer 2: this note is in an odd location. is it out of order?

> * Hive, which provisions and manages self-managed OpenShift clusters to the hub.

Reviewer 3: Suggest to expand to the following:
Hive, which provisions and performs initial configurations for self-managed OpenShift clusters to the hub.

> * Zero touch provisioning (ZTP), which provides full automation for cluster creation on bare metal and vSphere with minimal intervention. ZTP also offers a GitOps workflow for managing deployments.

Reviewer 2: this could link to the ZTP doc topic in OCP doc?

Reviewer 4: +1 to add a link to the ZTP docs. If the new ZTP docs here https://docs.openshift.com/container-platform/4.11/scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.html apply to MCE this should link to them

> * Support for hosted control planes for {product-title}.

Reviewer 2: would it be OK to include:
"(from the **HyperShift** project) somewhere on this page so that if people search for that term in the doc, it will land them here?"

Reviewer 3: Seems odd that we give a 1-liner on what Hosted control planes does (whereas the other bullets do).  Suggesting the following instead:

Hosted control planes provides a centralized control plane for OpenShift clusters which enables you to operate multiple clusters in a hyperscale manner. Hosted control planes is available on {product-title}.

> The operator is entitled with a Red Hat OpenShift support subscription and is delivered separately from the core OpenShift payload. To start to use mulicluster engine, you must deploy the OpenShift cluster and then install the operator. 

Reviewer: Can we point to the install docs here?

Author: Sure! I'll add the link now.

> The operator is entitled with a Red Hat OpenShift support subscription and is delivered separately from the core OpenShift payload. To start to use mulicluster engine, you must deploy the OpenShift cluster and then install the operator. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/multicluster_engine/multicluster_engine_overview#mce-install-intro[the installation documentation].

Author: Good catch, Chris!

[id="mce-additional-resources-ocp"]
== Additional resources

> For the complete documentation for multicluster engine, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/multicluster_engine/index[multicluster engine documentation], which is part of the product documentation for Red Hat Advanced Cluster Management. 

Reviewer 2: Should we also include a note to the effect of:
"If you are entitled to Red Hat Advanced Cluster Management you can deploy the operator and get full management capabilities.  See the Red Hat Advanced Cluster Management install doc topic <link>"

:_content-type: ASSEMBLY
[id="mce-overview-ocp"]
= About multicluster engine for Kubernetes operator
include::_attributes/attributes-openshift-dedicated.adoc[]
:context: mce-overview-ocp

toc::[]

> One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator (MCE). The operator delivers full lifecycle capabilities for managed OpenShift clusters and partial lifecycle management for other Kubernetes distributions. It is available in two ways:

Reviewer 5: ```suggestion
One of the challenges of scaling Kubernetes environments is managing the lifecycle of a growing fleet. To meet that challenge, you can use multicluster engine for Kubernetes operator (MCE). The operator delivers full lifecycle capabilities for managed {product-title} clusters and partial lifecycle management for other Kubernetes distributions. It is available in two ways:
```
Replaced "openshift" with the attribute variable.

> * As a standalone operator that you install as part of your OpenShift or Red Hat OpenShift Kubernetes Engine subscription

Reviewer 5: ```suggestion
* As a standalone operator that you install as part of your {product-title} or {oke} subscription
```
Replaced the product names with the attribute variables.

* As part of link:https://access.redhat.com/products/red-hat-advanced-cluster-management-for-kubernetes[Red Hat Advanced Cluster Management for Kubernetes]

[id="mce-on-ocp"]
== Cluster management with multicluster engine on {product-title}

When you enable multicluster engine on {product-title}, you gain the following capabilities:

> * xref:control-plane.adoc#hosted-control-planes-overview_control-plane[Hosted control planes], which is based on the HyperShift project. With hosted control planes, you have a centralized control plane for OpenShift clusters so that you can operate clusters in a hyperscale manner.

Reviewer 5: I'd recommend either "Hosted control planes, which are based on..." or "Hosted control planes, which is a feature based on..." 

Also, is "hyperscale" a well known technical term? It's new to me and I'm wondering if you should either use a more generic word or define "hyperscale".  Just a couple of wording suggestions and replacing "openshift" with the variable.

```suggestion
* xref:control-plane.adoc#hosted-control-planes-overview_control-plane[Hosted control planes], which are based on the HyperShift project. Centralized hosted control planes let you operate {product-title} clusters in a hyperscale manner.
```

Author: Good feedback here! 

I think I'll use "Hosted control planes, which is a feature that is based on the HyperShift project." This is one of those tricky feature names  because technically it's singular, but it sounds like we're talking about multiple things. 

I had the same thought about "hyperscale" being not well known enough, but I did some searching, and it looks like it's a term that's being adopted in the industry. I don't find it mentioned in the ISG, SSG, or the OpenShift glossary, but then again, those sources don't define every technical term. I'm tempted to leave it for now unless we get more feedback that it's unclear. 

I like your suggested rewrite, but I might tweak it a bit to avoid using the phrase "hosted control planes let you...". The ISG warns against phrases like "lets you..." or "allows you..." because by using those phrases, we're assigning human characteristics to inanimate objects (see [Anthropomorphism](https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism)). 

I'm thinking of changing it to something like, "Hosted control planes, which is a feature that is based on the HyperShift project. With a centralized hosted control plane, you can operate {product-title} clusters in a hyperscale manner."

Reviewer 5: This sounds great to me!

> * Hive, which provisions self-managed OpenShift clusters to the hub and completes the initial configurations for those clusters.

Reviewer 5: ```suggestion
* Hive, which provisions self-managed {product-title} clusters to the hub and completes the initial configurations for those clusters.
```

I'm not familiar with this area of the documentation - is "the hub" a well known term here or should it be described more specifically here?

Author: I believe "the hub" will be understood by customers who are familiar with cluster management.

* klusterlet agent, which registers managed clusters to the hub.

> * Infrastructure Operator, which manages the deployment of the Assisted Service to orchestrate on-premise bare metal and vSphere installations of OpenShift, such as SNO on bare metal. The Infrastructure Operator includes link:https://docs.openshift.com/container-platform/4.11/scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.html[zero touch provisioning (ZTP)], which fully automates cluster creation on bare metal and vSphere provisioning with GitOps workflows to manage deployments and configuration changes.

Reviewer 5: ```suggestion
* Infrastructure Operator, which manages the deployment of the Assisted Service to orchestrate on-premise bare metal and vSphere installations of {product-title}, such as SNO on bare metal. The Infrastructure Operator includes link:https://docs.openshift.com/container-platform/4.11/scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.html[zero touch provisioning (ZTP)], which fully automates cluster creation on bare metal and vSphere provisioning with GitOps workflows to manage deployments and configuration changes.
```

Should this link be an xref?

Author: @bscott-rh  I had the same question, and I initially tried to make it an xref, but I couldn't get it to work. Then, I noticed that I couldn't find that file anywhere in the openshift-docs repo, so I gave up and used a link. I can play around with it again this morning to see if I can get an xref to work.

* Open cluster management, which provides resources to manage Kubernetes clusters.

> The multicluster engine is included with your {product-title} support subscription and is delivered separately from the core payload. To start to use multicluster engine, you deploy the {product-title} cluster and then install the operator. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/multicluster_engine/multicluster_engine_overview#mce-install-intro[the installation documentation].

Reviewer 6: will we be refreshing this to the latest doc when 2.7+ comes out?

Author: Yes. The OCP docs go live about 2 weeks before ACM 2.7 is released, so I have a to-do for myself to update this link after ACM 2.7 is live.

> The multicluster engine is entitled with your Red Hat OpenShift support subscription and is delivered separately from the core OpenShift payload. To start to use multicluster engine, you deploy the OpenShift cluster and then install the operator. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/multicluster_engine/multicluster_engine_overview#mce-install-intro[the installation documentation].

Reviewer 5: ```suggestion
The multicluster engine is included? with your Red Hat {product-title} support subscription and is delivered separately from the core {product-title} payload. To use multicluster engine, you deploy the {product-title} cluster and then install the operator. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.6/html/multicluster_engine/multicluster_engine_overview#mce-install-intro[the installation documentation].
```
Brief wording suggestion.

[id="mce-on-rhacm"]
== Cluster management with Red Hat Advanced Cluster Management

> If you need cluster management capabilities beyond what {product-title} with multicluster engine can provide, consider Red Hat Advanced Cluster Management. The multicluster engine is an integral part of Red Hat Cluster Management and is enabled by default. 

Author: Good catch!

> If you need cluster management capabilities beyond what {product-title} with multicluster engine can provide, consider Red Hat Advanced Cluster Management. The multicluster engine is an integral part of Red Hat Cluster Management and is enabled by default. 

Author: Fixed

> If you need cluster management capabilities beyond what {product-title} with multicluster engine can provide, consider Red Hat Advanced Cluster Management. The multicluster engine is an integral part of Red Hat Advanced Cluster Management and is enabled by default. 

Reviewer 6: should we link to the ACM landing page or doc here?

Author: I linked to the ACM landing page at the first mention of ACM near the beginning of the topic.
